### ISSUE TITLE
Improve handling of narrows (views) with no messages

### ISSUE THREAD

**Issue by neiljp** [zulip/zulip-terminal#1506]
### Bug description

When selecting a new narrow (view) to show, via a hotkey, a button in either side panel, or using the message search (via `/`), the message focus changes to a message in the resulting view.

When the message list for that narrow/view is **empty**, ie. no messages are found, there is no visible message to gain focus, and the UI behaves strangely.

This can occur for many narrows/views, including:
- streams (channels)
  - #607 
  - #259
- searches which result in no results
  - #580
- direct messages with a specific user (eg. if not messaged them before)
  - #1480
- starred messages
  - ([#zulip-terminal > keyboard movement stops with no messages starred section](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/keyboard.20movement.20stops.20with.20no.20messages.20starred.20section))
- topics (difficult, but possible via exiting compose to a new topic)
- mentions (if never mentioned?)
- all direct messages (?)

This results in
- no obvious focused UI element (#607)
- the 'Current message recipients' not updating (remaining that of the previously focused message)
- the message narrow potentially not updating (#1480, #259)

It is possible to navigate out of this state by moving sideways and down (#580), or using one of the 'global' hotkeys such as `w` or `q`, P/#/f (though *not* `a`). However, these are workarounds.

This has been discussed variously in #zulip-terminal, and in at least the above linked issues, as well as early work on it in #278.

### What did you expect to happen?

Certainly:
- Empty the current message recipients content, or hide that section entirely (as per another issue)
- Ensure the narrow title is properly updated (eg. starred, etc.)

In addition:

1) Unlike the web app, we expect the focus to move somewhere after a new narrow is requested, so a placeholder in the message list may be the easiest short-term solution to use.

This was worked on historically in #278 as a fix for #259, where a 'fake message' was used as a placeholder. However, as demonstrated by the more complex logic we now support for searching/filtering the side panels, it should be reasonable to support a custom element for 'no results found' to be shown instead.

Even so, that solution would need to support various cases where messages 'arrive' in that empty narrow, and the placeholder is replaced by them.

2) Alternatively, focus could be retained in the previous location, if the resulting narrow is found to be empty. This would be convenient for message searches, where an empty set of results might be reasonably expected by a user to leave the focus in the search box to adjust the search.

3) Some combination of (1) and (2), since
- a placeholder shows that the narrow is empty (no messages)
- a placeholder can be focused, allowing easy movement horizontally across the app (left/right), such as from left panel to right panel.

### Notes

- I'm closing many of the linked issues, since they are essentially duplicates, ie. examples of the symptoms from the underlying problem.
- This could unblock other features, such as #1049

**Comment by Niloth-p** [comment]
Adding a few more related aspects.

- Currently, it doesn't seem to be possible to open ZT with new bots (bots without any prior messages).
Error message: `IndexError: Can't set focus, ListBox is empty`
This must be because they have no messages and all their narrows are empty (A user would get a message from Welcome Bot and may be subscribed to some channels by default, a bot would not be).

- I've noticed that the recipient bar shows "DONT HIDE" when ZT launches, for some users and bots (mostly just new accounts?), but not in every case.

**Comment by sahith-ch** [comment]
Zulip claim

**Comment by sahith-ch** [comment]
@zulipbot claim

**Comment by zulipbot** [comment]
Welcome to Zulip, @sahith-ch! We just sent you an invite to collaborate on this repository at https://github.com/zulip/zulip-terminal/invitations. Please accept this invite in order to claim this issue and begin a fun, rewarding experience contributing to Zulip!

Here's some tips to get you off to a good start:

- Join me on the [Zulip developers' server](https://chat.zulip.org), to get help, chat about this issue, and meet the other developers.
- [Unwatch this repository](https://help.github.com/articles/unwatching-repositories/), so that you don't get 100 emails a day.

As you work on this issue, you'll also want to refer to the [Zulip code contribution guide](https://zulip.readthedocs.io/en/latest/contributing/index.html), as well as the rest of the developer documentation on that site.

See you on the other side (that is, the pull request side)!

**Comment by zulipbot** [comment]
@sahith-ch You have been unassigned from this issue because you have not made any updates for over 14 days. Please feel free to reclaim the issue if you decide to pick up again. Thanks!

**Comment by arishazakry** [comment]
@zulipbot claim

**Comment by zulipbot** [comment]
Welcome to Zulip, @arishazakry! By picking up a Zulip issue, you're promising to do your best to follow [Zulip's contributing guidelines](https://zulip.readthedocs.io/en/latest/contributing/index.html). Neglecting to do so wastes maintainer time, and may result in being banned from contributing to Zulip.

To get started:

1. Work through all the "Prior to picking up your first issue" sections of the [Zulip contributor guide](https://zulip.readthedocs.io/en/latest/contributing/index.html).
2. Post a comment here describing your plan for how to approach this issue. Before you claim an issue, you should be confident that you will be able to tackle it effectively.
3. Accept the invite to collaborate on this repository at https://github.com/zulip/zulip-terminal/invitations. To avoid filling your inbox, we recommend [unwatching this repository](https://help.github.com/articles/unwatching-repositories/).

The issue will be assigned to you when all three steps have been completed. If you get stuck, you can always `@zulipbot abandon` this issue, and pick a different one.

If someone else claims this issue while you're working through the checklist, no worries! You can contribute by providing feedback on their pull request. If you’ve made good progress in understanding part of the codebase, you can also find another “help wanted” issue in the same area to work on.

**Comment by zulipbot** [comment]
@arishazakry You have been unassigned from this issue because you have not made any updates for over 14 days. Please feel free to reclaim the issue if you decide to pick up again. Thanks!

**Comment by arishazakry** [comment]
@zulipbot claim

**Comment by zulipbot** [comment]
@arishazakry You have been unassigned from this issue because you have not made any updates for over 14 days. Please feel free to reclaim the issue if you decide to pick up again. Thanks!
